src/cpp/ext/csm/metadata_exchange.cc

```diff
@@ -301,8 +309,8 @@ class MeshLabelsIterable : public LabelsIterable {
 
   const std::vector<std::pair<absl::string_view, std::string>>& local_labels_;
   // Holds either the metadata slice or the decoded proto struct.
-  mutable absl::variant<grpc_core::Slice, StructPb> metadata_;
-  mutable GcpResourceType type_;
+  mutable absl::variant<absl::optional<grpc_core::Slice>, StructPb> metadata_;
```

Comment:
Instead of putting `absl::optional<>` inside `absl::variant<>`, consider using `absl::monostate` as another possible type inside the `variant<>`.  I think that should give a more compact memory representation, although it may not be worth if it it makes the code more complicated.

Reply:
I can also simply say that an empty slice is not a valid value for the metadata, and that makes it more simple

Reply:
Yeah, that works too.